src/main/scala/rps/Game.scala

```diff
@@ -0,0 +1,43 @@
+package rps
+import java.util.Scanner
+import java.util.InputMismatchException
+
+object Game {
+  def moveRepr(move: Int) : String = {
+    move match {
+      case 0 => return "Rock"
+      case 1 => return "Paper"
+      case 2 => return "Scissors"
+      case _ => return "Invalid move!"
+    }
+  }
+
+  def detectGameEndResult(playerMove: Int, computerMove: Int) : String = {
+    (playerMove, computerMove) match {
+      case (x, y) if x == y => return "Draw!"
+      case (0, 1) => return "I Won!"
+      case (0, 2) => return "You Won!"
+      case (1, 0) => return "You won!"
+      case (1, 2) => return "I won!"
+      case (2, 0) => return "I won!"
+      case (2, 1) => return "You won!"
+      case (_, _) => return "Invalid Round!!"
+    }
+  }
+
+  def play() : Unit = {
+    var scanner = new Scanner(System.in)
+    val r = scala.util.Random
+    print("Choose your move between Rock(0), Paper(1) or Scissors(2): ")
+
+    val myPlay = r.nextInt(3)
+    val userPlay = try {
+      scanner.nextInt()
```

Comment:
this is fine, but you can replace it with 

```scala
import scala.io.StdIn.readLine
// ...
val userPlay = readLine("Choose your move between Rock(0), Paper(1) or Scissors(2): ")
```

which is much shorter

Reply:
oh, I see you used it in the Main, any reason why you used a Scanner here?

Reply:
Mmm, just because I wanted to get an Int for the userPlay instead of a String, which I assumed that's what Readline would give me, probably also with the \n which I didn't want to take care of... 
I'll replace it with readLine.

Reply:
But I guess just declaring:

    val userMove: Int = readLine 

is probably enough to get some casting done.

Reply:
mm no, it shouldn't compile. `readLine` explicitly returns a `String`